src/test/kotlin/com/exactpro/th2/conn/dirty/fix/TestStrategies.kt

```diff
@@ -0,0 +1,514 @@
+                    duration = Duration.of(2, ChronoUnit.SECONDS),
+                    cleanUpDuration = Duration.of(2, ChronoUnit.SECONDS),
+                    missIncomingMessagesConfiguration = MissMessageConfiguration(3)
+                ),
+            )
+        )) { msg, mtd, mode ->
+            messages.add(Triple(msg, mtd, mode))
+        }
+
+        val channel = testContext.channel
+        val handler = testContext.fixHandler
+
+        val captor = argumentCaptor<ByteBuf> {  }
+
+        verify(channel, timeout(2100)).open()
+        verify(channel).send(any(), any(), anyOrNull(), any()) // Logon
+        clearInvocations(channel)
+
+        handler.onIncoming(channel, businessMessage(3))
+        handler.onIncoming(channel, businessMessage(4))
+        handler.onIncoming(channel, businessMessage(5))
+
+        verify(channel, timeout(2100)).open()
+        verify(channel).send(any(), any(), anyOrNull(), any()) // Logon
+        clearInvocations(channel)
+        handler.onIncoming(channel, businessMessage(6))
+
+        verify(channel, times(1)).send(captor.capture(), any(), anyOrNull(), any())
+
+        channel.close()
+
+        captor.firstValue.apply { contains("35=2") }
+    }
+
+    @Test
+    fun testTransformLogonMessagesStrategy() {
+        val messages = mutableListOf<Triple<ByteBuf, Map<String, String>, IChannel.SendMode>>()
+        val testContext = createTestContext(BrokenConnConfiguration(
+            SchedulerType.CONSECUTIVE,
+            listOf(
+                RuleConfiguration(RuleType.DEFAULT, duration = Duration.of(2, ChronoUnit.SECONDS), cleanUpDuration = Duration.of(0, ChronoUnit.SECONDS)),
+                RuleConfiguration(
+                    RuleType.TRANSFORM_LOGON,
+                    duration = Duration.of(4, ChronoUnit.SECONDS),
+                    cleanUpDuration = Duration.of(2, ChronoUnit.SECONDS),
+                    transformMessageConfiguration = TransformMessageConfiguration("test", "A", numberOfTimesToTransform = 2)
+                ),
+            )
+        )) { msg, mode, mtd ->
+            messages.add(Triple(msg, mode, mtd))
+        }
+
+        val channel = testContext.channel
+
+        messages.clear()
+        verify(channel, timeout(6200).times(3)).open()
+
+        // start
+        verify(channel, times(5)).send(any(), any(), anyOrNull(), any())
+
+        messages[0].apply { assertTrue { first.contains("35=5") } }
+        messages[1].apply {
+            assertTrue { first.contains("35=A") }
+            assertEquals(second[RULE_NAME_PROPERTY], "test")
+        }
+        messages[2].apply { assertTrue { first.contains("35=5") } }
+        messages[3].apply {
+            assertTrue { first.contains("35=A") }
+            assertEquals(second[RULE_NAME_PROPERTY], "test")
+        }
+        messages[4].apply {
+            assertTrue { first.contains("35=A") }
+            assertEquals(second[RULE_NAME_PROPERTY], null)
+        }
```

Comment:
It is not really clear what we try to test here. Maybe we should add additional assertions regarding the content of the message (what was changed)